tests/test_validators.py

```diff
@@ -103,3 +106,37 @@ def test_validate_type_bad(value: Any, expected: type, have: type):
         ValidationError, match=re.escape(f"Expected {expected}, got {have}")
     ):
         validate_type(value, expected)
+
+
+def test_validate_variants_json():
+    json_file = pathlib.Path("tests/artifacts/variants.json")
+    assert json_file.exists(), "Expected JSON file does not exist"
+
+    # Read the Variants JSON file
+    with json_file.open() as f:
+        data = json.load(f)
+
+    validate_variants_json(data)
+
+
+def test_validate_variants_json_empty():
+    validate_variants_json({"variants": {}})
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {},
```

Comment:
Let's cover some pathological cases too.

```suggestion
        {},
        {"variants": ["abcd1234"]},
```